src/cascadia/TerminalControl/HwndTerminal.cpp

```diff
@@ -242,6 +315,11 @@ try
 {
     // As a rule, detach resources from the Terminal before shutting them down.
     // This ensures that teardown is reentrant.
+    if (_tsfHandle)
+    {
+        _tsfHandle.Unfocus(&_tsfDataProvider);

```

Comment:
I believe you don't need to unfocus during teardown. Re-reading its implementation shows that it is primarily concerned with clearing the pending compositions out of the renderer.